src/restore.js

```diff
@@ -25,156 +26,64 @@ const visit = require("unist-util-visit");
  *   // returns "Markdown containing [a modified link](http://example.com) that has been redacted"
  *   unified().use([
  *     parse,               // use the standard parser
- *     restore(sourceTree), // use this extension with the source content above
+ *     findRestorations, // find Restoration nodes
+ *     restore(sourceTree, restorationMethods), // use with a tree of the source content and the defined restoration methods
  *     stringify            // output back to markdown
  *   ]).processSync(redacted);
  *
  * @see https://github.com/remarkjs/remark/tree/remark-parse%405.0.0/packages/remark-parse#extending-the-parser
  * @see render
  */
-module.exports = function restoreRedactions(sourceTree) {
+module.exports = function restoreRedactions(sourceTree, restorationMethods) {
   // First, walk the source tree and find all redacted nodes.
   const redactions = [];
   visit(sourceTree, ["inlineRedaction", "blockRedaction"], function(node) {
     redactions.push(node);
   });
 
-  // then return an extension to the parser that can consume the data from these
-  // redacted nodes when it encounters a redaction
-  return function () {
-    if (!this.Parser) {
-      return;
+  function blockVisitor(node, index, parent) {
+    const redactedData = redactions[node.redactionIndex];
+    if (!redactedData || redactedData.type !== "blockRedaction") {
+      const nodes = [
+        {type: "rawtext", value: "[][" + node.redactionIndex + "]"},
+        ...node.children,
+        {type: "rawtext", value: "[/][" + node.redactionIndex + "]"}
+      ];
```

Comment:
why do this instead of leaving it as a `blockRestoration` node?

Reply:
Changed!